src/sidebar/components/test/PaginationNavigation-test.js

```diff
@@ -0,0 +1,159 @@
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+
+import PaginationNavigation, { $imports } from '../PaginationNavigation';
+
+describe('PaginationNavigation', () => {
+  let fakeOnChangePage;
+  let fakePageNumberOptions;
+
+  let container;
+
+  const findButton = (wrapper, title) =>
+    wrapper.find('button').filterWhere(n => n.props().title === title);
+
+  const createComponent = (props = {}) => {
+    return mount(
+      <PaginationNavigation
+        currentPage={1}
+        onChangePage={fakeOnChangePage}
+        totalPages={10}
+        {...props}
+      />,
+      { attachTo: container }
+    );
+  };
+
+  beforeEach(() => {
+    // This extra element is necessary to test automatic `focus`-ing
+    // of the component's `input` element
+    container = document.createElement('div');
+    document.body.appendChild(container);
+
+    fakeOnChangePage = sinon.stub();
+    fakePageNumberOptions = sinon.stub().returns([1, 2, 3, 4, null, 10]);
+
+    $imports.$mock({
+      '../util/pagination': { pageNumberOptions: fakePageNumberOptions },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  describe('prev button', () => {
+    it('should render a prev button when there are previous pages to show', () => {
+      const wrapper = createComponent({ currentPage: 2 });
+
+      const button = findButton(wrapper, 'Go to previous page');
+
+      assert.isTrue(button.exists());
+    });
+
+    it('should not render a prev button if there are no previous pages to show', () => {
+      const wrapper = createComponent({ currentPage: 1 });
+
+      const button = findButton(wrapper, 'Go to previous page');
+
+      assert.isFalse(button.exists());
+    });
+
+    it('should invoke the onChangePage callback when clicked', () => {
+      const wrapper = createComponent({ currentPage: 2 });
+
+      const button = findButton(wrapper, 'Go to previous page');
+
+      button.simulate('click');
+
+      assert.calledWith(fakeOnChangePage, 1);
+    });
+
+    it('should remove focus from button after clicked', () => {
+      const wrapper = createComponent({ currentPage: 2 });
+
+      const button = findButton(wrapper, 'Go to previous page');
+      const buttonEl = button.getDOMNode();
+      const blurSpy = sinon.spy(buttonEl, 'blur');
```

Comment:
If you're using `sinon.spy` to observe the `blur` call, does this component need still to be rendered into a connected DOM container (ie. one that is a descendant of `document.body`)?

Reply:
Ah! Probably not, thanks!